### PR TITLE
Prevent JS errors on skipped map page

### PR DESCRIPTION
**Pull request by dracos**
Fixes https://github.com/mysociety/fixmystreet/issues/2543
Also factored together some shared styles and stylemaps (given one error to fix the above was caused by pin_prefix).

### CODE REVIEW

File: web/cobrands/fixmystreet/assets.js [hunk 1]
```diff
@@ -505,15 +480,56 @@ function get_fault_stylemap() {
...
+        popupYOffset: -40,
+        graphicOpacity: 1.0
+    }),
+
     selectedFeature: function() {
         return selected_feature;
     },
 
     add: function(default_options, options) {
+        if (!fixmystreet.map_type) {
```
Comment (davea) [inline web/cobrands/fixmystreet/assets.js]: This breaks assets on certain cobrands (I've tried BANES, Hounslow, Bristol, Lincolnshire, Bexley. Works OK on Bucks though). It looks as though `map_type` isn't set at this point so no asset layers get added to the map.

File: web/js/map-OpenLayers.js [hunk 2]
```diff
@@ -791,6 +794,10 @@ $.extend(fixmystreet.utils, {
 
     $(function(){
 
+        if (!fixmystreet.map_type) {
```
Comment (davea) [inline web/js/map-OpenLayers.js]: This test fails on Bexley and Lincolnshire and breaks the map for some reason.

File: web/cobrands/fixmystreet/assets.js (hunk 1 above)
Comment (dracos) [reply web/cobrands/fixmystreet/assets.js]: Okay, turns out map_type isn't always defined, I thought it was; I've switched to checking for the presence of the map div.

File: web/js/map-OpenLayers.js (hunk 2 above)
Comment (dracos) [reply web/js/map-OpenLayers.js]: Okay, turns out map_type isn't always defined, I thought it was; I've switched to checking for the presence of the map div.

File: web/cobrands/fixmystreet/assets.js (hunk 1 above)
Comment (dracos) [reply web/cobrands/fixmystreet/assets.js]: Plus rewriting the code to always make sure map.js loads before assets.js. But then still back to the blank issue as elsewhere.

**Pull request merged**